Tools/cases_generator/tier2_generator.py

```diff
@@ -167,7 +167,7 @@ def oparg(
         return True
 
 
-def write_uop(uop: Uop, emitter: Emitter, stack: Stack) -> Stack:
+def write_uop(uop: Uop, emitter: Emitter, stack: Stack, inst: Instruction | None) -> Stack:
```

Comment:
Maybe pass the size, not the instruction. A uop can occur in many instructions, so picking one is arbitrary.
However, if it uses `INSTRUCTION_SIZE`, than all instructions must be the same size.

Reply:
I added the check for the instruction size, which actually pointed out one place where we cannot use `INSTRUCTION_SIZE` instead of `next_instr - this_instr`:

https://github.com/python/cpython/blob/d8c864816121547338efa43c56e3f75ead98a924/Python/bytecodes.c#L4799